### PR TITLE
Fix cache sanity checking code by moving functions to wider scope

### PR DESCRIPTION
**Pull request by jhendersonHDF**

### CODE REVIEW

File: src/H5C.c [hunk 1]
```diff
@@ -7788,51 +7776,48 @@ H5C__validate_lru_list(H5C_t *cache_ptr)
         (cache_ptr->LRU_head_ptr != cache_ptr->LRU_tail_ptr))
         HGOTO_ERROR(H5E_CACHE, H5E_SYSTEM, FAIL, "Check 1 failed")
 
-    if (cache_ptr->LRU_list_len < 0)
```
Comment (jhendersonHDF) [inline src/H5C.c:7791]: Warning generated by checking uint32_t for < 0

File: src/H5C.c [hunk 2]
```diff
@@ -7865,54 +7850,51 @@ H5C__validate_pinned_entry_list(H5C_t *cache_ptr)
         (cache_ptr->pel_head_ptr != cache_ptr->pel_tail_ptr))
         HGOTO_ERROR(H5E_CACHE, H5E_SYSTEM, FAIL, "Check 1 failed")
 
-    if (cache_ptr->pel_len < 0)
```
Comment (jhendersonHDF) [inline src/H5C.c:7868]: Warning generated by checking uint32_t for < 0

File: src/H5C.c [hunk 3]
```diff
@@ -7945,54 +7927,51 @@ H5C__validate_protected_entry_list(H5C_t *cache_ptr)
         (cache_ptr->pl_head_ptr != cache_ptr->pl_tail_ptr))
         HGOTO_ERROR(H5E_CACHE, H5E_SYSTEM, FAIL, "Check 1 failed")
 
-    if (cache_ptr->pl_len < 0)
```
Comment (jhendersonHDF) [inline src/H5C.c:7948]: Warning generated by checking uint32_t for < 0

**Review by gnuoyd — APPROVED** [review]
This is fine.

**Review by lrknox — APPROVED** [review]

**Pull request merged**